### PR TITLE
fix: net/http incorrect offsets

### LINKED ISSUE #1204: The offset for golang.org/x/net/http2.FrameHeader.StreamID in the offset_results.json file was incorrectly generated for version 0.8.0
### Describe the bug

The offset for golang.org/x/net/http2.FrameHeader.StreamID  in [the offset_results.json](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/main/internal/pkg/inject/offset_results.json#1546) for version 0.8.0 should be the same as other versions.

[the source for v0.7.0](https://pkg.go.dev/golang.org/x/net@v0.7.0/http2#FrameHeader)
[the source for v0.8.0](https://pkg.go.dev/golang.org/x/net@v0.8.0/http2#FrameHeader)
[the source for v0.9.0](https://pkg.go.dev/golang.org/x/net@v0.9.0/http2#FrameHeader)

### PR DESCRIPTION
**Pull request by damemi**
Fixes #1204 

Not sure if this is worth backporting but if so I can take that

### CODE REVIEW

File: internal/pkg/inject/offset_results.json [hunk 1]
```diff
@@ -1679,7 +1669,8 @@
                       "1.7.2",
                       "1.7.3",
                       "1.7.4",
-                      "1.7.5"
+                      "1.7.5",
+                      "1.69.0-dev"
```
Comment (RonFed) [inline internal/pkg/inject/offset_results.json:1673]: That looks suspicious

Comment (damemi) [reply internal/pkg/inject/offset_results.json:1673]: Yeah, odd... this version was also wholly removed from another section below https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1206/files#diff-15a322f23067198dc3f6baea35decb97f02a703fe637d6191e9871c5bf43d564L3793-L3796

Comment (MrAlias) [reply internal/pkg/inject/offset_results.json:1673]: Hmm, yeah, this seems wrong. `google.golang.org/grpc@v1.69.0-dev` depends on `google.golang.org/genproto/googleapis/rpc@v0.0.0-20240903143218-8af14fe29dc1`:

https://github.com/grpc/grpc-go/blob/4544b8a4cfe9bb4882ec3591631b83dac7434805/go.mod#L16

That version of that package definitely has a `Status.Code` field:

https://pkg.go.dev/google.golang.org/genproto/googleapis/rpc@v0.0.0-20240903143218-8af14fe29dc1/status#Status

How was this generated?

Comment (damemi) [reply internal/pkg/inject/offset_results.json:1673]: I generated this by deleting my local `.tools/` and `offset_results.json` then running `make docker-offsets`. Trying to see if I can get something more reasonable by updating the Go version in the [image](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/741985e9b79cb1fd49ffcd9935cbe16288f53e8a/Makefile#L122) and [tools](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/741985e9b79cb1fd49ffcd9935cbe16288f53e8a/internal/tools/go.mod#L3)

Comment (damemi) [reply internal/pkg/inject/offset_results.json:1673]: Couldn't get any difference, but I checked back and it was actually the offset for [`http2client.nextID`](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/bcfb963ffdc0ada3506ab54fd3e22d9bbc6fbfac/internal/pkg/inject/offset_results.json#L3554-L3558) that was removed in this PR for that version: https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1206/files#diff-15a322f23067198dc3f6baea35decb97f02a703fe637d6191e9871c5bf43d564L3791-L3796

And it was added to the `null` offset for that field on this line: https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1206/files#diff-15a322f23067198dc3f6baea35decb97f02a703fe637d6191e9871c5bf43d564R3598-R3599

Looking more at this PR, it looks like it removed all the fields in `1.69.0-dev` from having an offset to being `null`, so something's up there

**Comment by MrAlias** [comment]
Tracking issue in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1226. Closing.

**Pull request closed without merging**